common/src/main/java/org/vivecraft/client/gui/widgets/SettingsList.java

```diff
@@ -0,0 +1,230 @@
+package org.vivecraft.client.gui.widgets;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.blaze3d.vertex.PoseStack;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.ComponentPath;
+import net.minecraft.client.gui.components.*;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.client.gui.narration.NarratedElementType;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.navigation.FocusNavigationEvent;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import org.jetbrains.annotations.Nullable;
+import org.vivecraft.client.gui.settings.GuiListValueEditScreen;
+import org.vivecraft.server.config.ConfigBuilder;
+
+public class SettingsList extends ContainerObjectSelectionList<SettingsList.BaseEntry> {
+    final Screen parent;
+    int maxNameWidth;
+
+    public SettingsList(Screen parent, Minecraft minecraft, List<SettingsList.BaseEntry> entries) {
+        super(minecraft, parent.width + 45, parent.height, 20, parent.height - 32, 20);
+        this.parent = parent;
+        for (SettingsList.BaseEntry entry : entries) {
+            int i;
+            if ((i = minecraft.font.width(entry.name)) > this.maxNameWidth) {
+                this.maxNameWidth = i;
+            }
+            this.addEntry(entry);
+        }
+    }
+
+    @Override
+    protected int getScrollbarPosition() {
+        return super.getScrollbarPosition() + 8;
+    }
+
+    @Override
+    public int getRowWidth() {
+        return super.getRowWidth() + 32;
+    }
+
+    public static BaseEntry ConfigToEntry(ConfigBuilder.ConfigValue configValue, Component name) {
```

Comment:
I'm not too big of a fan of this type of `if else`. Can we instead have the widget be part of the config? So each config has a `get/to/makeWidget` method. But this might start overloading the config class a bit.

Reply:
moved it to the config values with the latest commit